### ISSUE TITLE
Statusnotifier plugin doesn't work in Ubuntu/Unity

### ISSUE THREAD

**Issue by Alexey-Yakovenko** [Alexey-Yakovenko/deadbeef#1608]
Research of the issue was conducted as part of another issue investigation: #1606

The "Menu" property must be added with the help of dbusmenu-glib, possibly by converting from GtkMenu via dbusmenu-gtk.

**Comment by Alexey-Yakovenko** [comment]
Our options:

* A new plugin using libappindicator
    * PROS
        * Already does all that's needed on Ubuntu.
        * Separate libs are provided GTK2 and GTK3.
        * Seems to have stable ABI (so version 1), so don't need to care about dbusmenu-glib versions
    * CONS
        * The plugin would not provide the adequate experience on KDE, and would duplicate most of the statusnotifier plugin functionality.
        * Would be a problem to find out which of the two plugins need to be used, since we can't ditch the statusnotifier.
* Modify the statusnotifier plugin to use libdbusmenu-gtk, and split it into GTK2 and GTK3 versions.
    * PROS
        * One plugin for all desktops
        * Best UX on each desktop, with all supported features
    * CONS
        * Need to link to libdbusmenu-glib and libdbusmenu-gtk directly (no stable ABI, so version 4 in Ubuntu 16.04)
        * A risk that libdbusmenu-glib and friends have high glib version requirement, so couldn't be easily included in the cross-distro builds

**Comment by anewuser** [comment]
Thank you for being considerate of KDE users.

**Comment by Alexey-Yakovenko** [comment]
Update on that:

It seems like the state of status notification area is still up in the air, and not unified across Linux distributions.

There's no standard, there are no libraries which work, no documentation.

GTK doesn't support that feature at all, and GNOME position on that "we don't have status notification area".

Deadbeef 1.8 will ship without the new statusnotification plugin, and will keep using XEmbed / GtkStatusIcon. This feature will be fixed as soon as Linux desktop environments come to a single standard.

This issue gets converted into enhancement instead of being a bug.

**Comment by arigit** [comment]
Running deadbeef 0.72 (the last stable release) on ubuntu 18.04. No status icon at all. The workaround of deleteting statusnotifier.so does not seem to apply, there is no such file in /usr/lib/deadbeef
Looking for to an actual stable release that solves the problem - incidentally the last release seems to be more than 2 years old

**Comment by Alexey-Yakovenko** [comment]
The situation with status notifications in Linux currently is very chaotic, and the deadbeef team doesn't have resources to support this feature at the moment. Unfortunately, even the next stable release won't fix this issue, but hopefully we can figure out something shortly after.

The Linux notification subsystems have changed more than once while we were still working on fixing our previous implementation, so we had to kill it.

**Comment by Alexey-Yakovenko** [comment]
Yeah, it's even listed on deadbeef's wiki: https://github.com/DeaDBeeF-Player/deadbeef/wiki/Plugins

**Comment by erm3nda** [comment]
I've installed it using `yay`. Doesn't work.
I moved from `/usr/lib/deadbeef/sni_gtk3.so` to `.local/lib/deadbeef` (I created it). Doesn't work and the registry dialog showed me that 2 `sni_gtk3` modules  attempted to load, so, it was loaded from the start, and was not a problem with the lib folder location.

I have no idea on how to make this work.

Linux Manjaro Gnome Edition.
Most (or all) other icons does work, not sure if this is a plugin problem or problem on my system.

**Comment by Alexey-Yakovenko** [comment]
@erm3nda 
There's no plugin called "sni_gtk3" in deadbeef. It's most likely a 3rd party plugin, made by someone else. You should seek help from the devs of that plugin instead of here.

**Comment by erm3nda** [comment]
@Alexey-Yakovenko Yeah, it's even listed on deadbeef's wiki: https://github.com/DeaDBeeF-Player/deadbeef/wiki/Plugins.

This is the plugin i am talking too, the plugin you where talking about on this thread.

**Comment by Alexey-Yakovenko** [comment]
@erm3nda 
Yeah, it's also written on the Plugins page:
> DISCLAIMER: DeaDBeeF Player developers don't have anything to do with these plugins. Use at your own risk.

**Comment by erm3nda** [comment]
Understood

**Comment by luzpaz** [comment]
Still relevant ?

**Comment by Oleksiy-Yakovenko** [comment]
yes, most open issues are open because they have been requested / reported but have not been resolved.
I hope you're not intending to ask in every one of them :D
